examples/auto-scaling-groups/init.pp

```diff
@@ -0,0 +1,71 @@
+ec2_securitygroup { 'test-sg':
+  ensure      => present,
+  description => 'group for testing autoscaling group',
+  region      => 'sa-east-1',
+}
+
+ec2_launchconfiguration { 'test-lc':
+  ensure          => present,
+  security_groups => ['test-sg'],
+  region          => 'sa-east-1',
+  image_id        => 'ami-67a60d7a',
+  instance_type   => 't1.micro',
+}
+
+ec2_autoscalinggroup { 'test-asg':
+  ensure               => present,
+  min_size             => 2,
+  max_size             => 4,
+  region               => 'sa-east-1',
+  launch_configuration => 'test-lc',
+  availability_zones   => ['sa-east-1b', 'sa-east-1a'],
+}
+
+ec2_scalingpolicy { 'scaleout':
+  ensure             => present,
+  auto_scaling_group => 'test-asg',
+  scaling_adjustment => 30,
+  adjustment_type    => 'PercentChangeInCapacity',
```

Comment:
I decided against adding validation to the types for a number of the properties for a few reasons (that are worth discussing here).

* The API is large, and evolving. So new options could be added by Amazon at any time. If we embed validation in the types we have to monitor that and release new versions. Not including validation minimises the maintenance overhead.
* The error messages from the API client are in my experience very good at doing the validation. For instance if you run this manifest with a random value you get the following:

```
Error: Could not set 'present' on ensure: Valid adjustment types are: [PercentChangeInCapacity, ChangeInCapacity, ExactCapacity]. at 30:/Users/garethr/Documents/puppetlabs-aws/tests/asg.pp
```

* It raises the barrier to entry for contributions from third parties, in some cases the additional code for validation rules and testing those rules would be greater than the provider adding the new functionality. The size of the API means I'd favour as low a barrier to entry as possible.

With regards to what the options mean, the module should require a solid understanding of the AWS primitives. It's not intended to be an abstraction (like say the Nginx module) but rather an alternative user interface. The example currently links to http://docs.aws.amazon.com/AutoScaling/latest/DeveloperGuide/WhatIsAutoScaling.html which covers the concepts.